crates/shared/src/test_utils.rs

```diff
@@ -0,0 +1,266 @@
+                            let error_msg = format!(
+                                "Key missing in expected JSON at {}",
+                                current_path.join("."),
+                            );
+                            current_path.pop();
+                            return Err(anyhow!(error_msg));
+                        }
+                        (None, None) => unreachable!(),
+                    }
+
+                    current_path.pop();
+                }
+                Ok(())
+            }
+            _ => {
+                if actual != expected {
+                    Err(anyhow!(
+                        "Mismatch at {}: {:?} != {:?}",
+                        current_path.join("."),
+                        actual,
+                        expected
+                    ))
+                } else {
+                    Ok(())
+                }
+            }
+        }
+    }
+
+    let mut current_path = vec![];
+    compare_jsons(actual, expected, exclude_paths, &mut current_path)
+}
+
+#[cfg(test)]
+mod tests {
+    use {super::*, maplit::hashset, serde_json::json};
+
+    #[test]
+    fn test_parse_field_paths() {
+        let paths = ["user.profile.name", "user.settings"];
+        let parsed_paths = parse_field_paths(&paths);
+        let expected_paths: HashSet<Vec<String>> = hashset! {
+            vec!["user".to_string(), "profile".to_string(), "name".to_string()],
+            vec!["user".to_string(), "settings".to_string()],
+        };
+        assert_eq!(parsed_paths, expected_paths)
+    }
+
+    #[test]
+    fn test_json_matches_excluding_no_exclusions() {
+        let json_a = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice"
+                }
+            }
+        });
+        let json_b = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice"
+                }
+            }
+        });
+        assert_json_matches!(json_a, json_b, [])
+    }
+
+    #[test]
+    fn test_json_matches_excluding_with_exclusions() {
+        let json_a = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice",
+                    "timestamp": "2021-01-01T12:00:00Z"
+                },
+                "enabled": true,
+            }
+        });
+        let json_b = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice",
+                    "timestamp": "2022-01-01T12:00:00Z"
+                },
+                "enabled": false,
+            }
+        });
+        assert_json_matches!(json_a, json_b, ["user.profile.timestamp", "user.enabled"])
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "JSON did not match with the exclusion of specified paths: Mismatch at \
```

Comment:
Might be nice to also print the 2 JSONs. In case there are a bunch of differences you could spot them all at once instead of having to rerun the test 5 times until you resolved all of them.

Reply:
> Might be nice to also print the 2 JSONs.

Added the jsons also.

> In case there are a bunch of differences you could spot them all at once instead of having to rerun the test 5 times until you resolved all of them.

That would probably require the use of an errors accumulator instead of returning a single error from the `json_matches_excluding` function.